test/simulator/test/org/apache/cassandra/simulator/test/AccordJournalSimulationTest.java

```diff
@@ -18,250 +18,359 @@
+
+                    Keyspace.setInitialized();
+
+                    CMSTestBase.CMSSut sut = new CMSTestBase.CMSSut(AtomicLongBackedProcessor::new,
+                            false,
+                            new FakeSchema(),
+                            new TokenPlacementModel.SimpleReplicationFactor(1));
+
+                    State.journal = new Journal<>("AccordJournal",
+                            new File("/journal"),
+                            new AccordSpec.JournalSpec(),
+                            new TestCallbacks(),
+                            new IdentityKeySerializer(),
+                            new IdentityValueSerializer());
+                }),
+                () -> check());
     }
 
-    private static void run()
+    public static void check()
     {
-        for (int i = 0; i < State.events; i++)
+        State.journal.start();
+        try
+        {
+            for (int i = 0; i < 100; i++)
+            {
+                int finalI = i;
+                State.executor.submit(() -> State.journal.asyncWrite("test" + finalI, "test" + finalI, Collections.singleton(1), null));
+            }
+
+            State.latch.await();
+
+            for (int i = 0; i < 100; i++)
+            {
+                System.out.println("Reading " + State.journal.readFirst("test" + i));
+                Assert.assertEquals(State.journal.readFirst("test" + i), "test" + i);
+            }
+        }
+        catch (Throwable e)
         {
-            int finalI = i;
-            State.executor.execute(() -> State.append(finalI));
+            e.printStackTrace();
         }
+        finally
+        {
+            State.journal.shutdown();
+        }
+    }
 
-        try
+    public static class TestCallbacks implements AsyncCallbacks<String, String>
+    {
+
+        @Override
+        public void onWrite(long segment, int position, int size, String key, String value, Object writeContext)
         {
-            State.eventsDurable.await();
-            State.logger.info("All events are durable done!");
+            State.latch.decrement();
         }
-        catch (InterruptedException e)
+
+        @Override
+        public void onWriteFailed(String key, String value, Object writeContext, Throwable cause)
         {
-            throw new AssertionError(e);
+            State.latch.decrement();
         }
 
-        if (!State.exceptions.isEmpty())
+        @Override
+        public void onFlush(long segment, int position)
         {
-            AssertionError error = new AssertionError("Exceptions found during test");
-            State.exceptions.forEach(error::addSuppressed);
-            throw error;
         }
 
-        State.journal.shutdown();
-        State.logger.info("Run complete");
+        @Override
+        public void onFlushFailed(Throwable cause)
+        {
+            new RuntimeException("Could not flush", cause).printStackTrace();
```

Comment:
this is a big issue as the test could pass and this *will* be ignored.  You should save this to `State` and assert this in the test.  Any failures is unexpected and should lead to a test fault